### ISSUE TITLE
Alls files getting note as changed

### ISSUE THREAD

**Issue by vzytoi** [lewis6991/gitsigns.nvim#399]
### Description

Hello, firstly thanks for the plugin. I tried to installed it following the readme file and when I restarted my neovim, all files of each files is getting noted as changed. See the linked screenshot.
Thank you for your interest in this issue

### Neovim version

NVIM v0.6.0-dev+205-g9695691ee Build type: RelWithDebInfo LuaJIT 2.1.0-beta3 Compilé par runneradmin@fv-az152-120  Features: -acl +iconv +tui See ":help feature-compile"           fichier vimrc système : "$VIM\sysinit.vim"                $VIM par défaut : "C:/Program Files/nvim/share/nvim"  Run :checkhealth for more info

### Operating system and version

Windows 10

### Steps to reproduce

No special step, I followed the readme

### Actual behavior

![image](https://user-images.githubusercontent.com/65293868/140383810-650cbf83-bcf5-459b-a9eb-bdb34384e03b.png)

### Gitsigns debug messages

`:Gitsigns debug_messages` :
```
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
run_job: git --no-pager --version
attach(1): Already attached
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
attach(1): Already attached
attach(1): Already attached
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
run_job: git --no-pager --version
attach(1): Already attached
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsDelete is already defined
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsAdd is already defined
setup_highlight: Highlight GitSignsChange is already defined
```
`:Gitsigns dump_cache` : 
```
{ {
    compare_file = "C:\\Users\\Cyprien\\AppData\\Local\\Temp\\nvimVj0Nwd\\1",
    compare_text = { "...",
      length = 0
    },
    file = "C:\\Users\\Cyprien\\AppData\\Local\\nvim\\minimal.lua",
    git_obj = {
      file = "C:\\Users\\Cyprien\\AppData\\Local\\nvim\\minimal.lua",
      has_conflicts = false,
      relpath = "minimal.lua",
      repo = {
        abbrev_head = "main",
        gitdir = "C:/Users/Cyprien/AppData/Local/nvim/.git",
        toplevel = "C:/Users/Cyprien/AppData/Local/nvim",
        username = "vzytoi"
      }
    },
    gitdir_watcher = <userdata 1>,
    hunks = { {
        added = {
          count = 42,
          lines = { "vim.cmd [[set runtimepath=$VIMRUNTIME]]\r", "vim.cmd [[set packpath=/tmp/nvim/site]]\r", "local package_root = '/tmp/nvim/site/pack'\r", "local install_path = package_root .. '/packer/start/packer.nvim'\r", "\r", "if vim.fn.isdirectory(install_path) ==
0 then\r", '  print("Installing Gitsigns and dependencies.")\r', "  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }\r", "end\r", "\r", "-- Load plugins\r", "require('packer').startup {\r", "  {\r", "    'wbthomason/packer.nvim',\r", "    {\r", "      'lewis6991/gitsigns.nvim',\r", "      requires = { 'nvim-lua/plenary.nvim' },\r", "    },\r", "    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "  },\r", "  config = {\r", "    package_root = package_root,\r", "    compile_path = install_path .. '/plugin/packer_compiled.lua',\r", "    display = { non_interactive = true },\r", "  },\r", "}\r", "\r", "_G.load_config = function()\r", "  require('gitsigns').setup{\r", "    debug_mode = true, -- You must add this to enable debug messages\r", "    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "  }\r", "\r", "  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "\r", "  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES\r", "  vim.cmd('Gitsigns debug_messages')\r", "end\r", "\r", "require('packer').sync()\r", "\r", 'vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]\r' },
          start = 1
        },
        dend = 42,
        head = "@@ -0 +1,42 @@",
        removed = {
          count = 0,
          lines = {},
          start = 0
        },
        start = 1,
        type = "add",
        vend = 42
      } },
    pending_signs = {},
    staged_diffs = {}
  } }
```

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'

if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Gitsigns and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end

-- Load plugins
require('packer').startup {
  {
    'wbthomason/packer.nvim',
    {
      'lewis6991/gitsigns.nvim',
      requires = { 'nvim-lua/plenary.nvim' },
    },
    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  },
  config = {
    package_root = package_root,
    compile_path = install_path .. '/plugin/packer_compiled.lua',
    display = { non_interactive = true },
  },
}

_G.load_config = function()
  require('gitsigns').setup{
    debug_mode = true, -- You must add this to enable debug messages
    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  }

  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE

  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES
  vim.cmd('Gitsigns debug_messages')
end

require('packer').sync()

vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]

```

**Comment by lewis6991** [comment]
You've not followed the issue template properly.

Will reopen when OP is updated with proper information. Let me know if you need help with any of the instructions in the template.

**Comment by vzytoi** [comment]
I think having change all that I could, the issue still occurs

**Comment by lewis6991** [comment]
The `debug_messages` output looks incomplete given that the cache looks fully populated. Can you double check the output?

**Comment by lewis6991** [comment]
Also I can see the problem in the cache:

```
    compare_text = { "...",
      length = 0
    },
```

Which says the output of `git show :0:minimal.lua` failed. This is probably due to slashes in some way.

**Comment by vzytoi** [comment]
Thank you for your feed back.
Here is a new output for `:Gitsigns debug_messages`:
```
setup_highlight: Deriving GitSignsDelete from DiffDelete
setup_highlight: Deriving GitSignsAdd from DiffAdd
setup_highlight: Deriving GitSignsChange from DiffChange
setup_highlight: Highlight GitSignsChange is already defined
setup_highlight: Highlight GitSignsDelete is already defined
run_job: git --no-pager --version
attach(1): Attaching (trigger=setup)
run_job: git --no-pager config user.name
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git ls-files --stage --others --exclude-standard C:\Users\Cyprien\AppData\Local\nvim\init.lua
watch_gitdir(1): Watching git dir
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:init.lua
update(1): updates: 1, jobs: 5
attach(2): Attaching (trigger=setup)
attach(2): Non-normal buffer
attach(3): Attaching (trigger=setup)
run_job: git --no-pager config user.name
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git ls-files --stage --others --exclude-standard C:\Users\Cyprien\AppData\Local\nvim\minimal.lua
watch_gitdir(3): Watching git dir
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua
update(3): updates: 2, jobs: 9
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
attach(3): Already attached
watcher_cb(1): Git dir update
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git ls-files --stage --others --exclude-standard C:\Users\Cyprien\AppData\Local\nvim\init.lua
watcher_cb(3): Git dir update
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
watcher_cb(1): File not changed
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git ls-files --stage --others --exclude-standard C:\Users\Cyprien\AppData\Local\nvim\minimal.lua
watcher_cb(3): File not changed
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua
update(3): updates: 3, jobs: 15
run_job: git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua
update(3): updates: 4, jobs: 16
```

**Comment by lewis6991** [comment]
So basically the command `git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua` isn't working for whatever reason. My guess is that it doesn't like the `/` and might require `\` instead.

I don't use or have access to a Windows machine, so you will need to debug this yourself. Let me know if you need any help.

**Comment by vzytoi** [comment]
I tried using `/` and `\` and I have this error : 
```
fatal: path 'minimal.lua' exists on disk, but not in the index
```

**Comment by vzytoi** [comment]
I solved this problem, using a [stackoverflow question](https://stackoverflow.com/questions/18755089/git-show-not-working-fatal-path-mypath-exists-on-disk-but-not-in-mybran) and this command now works perfectly find : `git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua` (it output the file content). But my issue still occures, I guess it wasn't the reason. If you have others idea it would be really nice, have a good day (:

**Comment by vzytoi** [comment]
Here is a new output for `:Gitsigns dump_cache`, I reallized that different slashes "format" are used : somethimes `/` and sometimes `\\` (like for compare_file). Could it be the reason? 
I checked the output for `cd C:\\Users\\Cyprien\\AppData\\Local\\nvim\\minimal.lua` and it does not work unlike `cd C:/Users/Cyprien/AppData/Local/nvim/minimal.lua` which works perfectly fine, so this is what let me think that using `\\` could make the issue happen
```
{ {
    compare_file = "C:\\Users\\Cyprien\\AppData\\Local\\Temp\\nvimKA4HoE\\1",
    compare_text = { "...",
      length = 0
    },
    file = "C:\\Users\\Cyprien\\AppData\\Local\\nvim\\minimal.lua",
    git_obj = {
      file = "C:\\Users\\Cyprien\\AppData\\Local\\nvim\\minimal.lua",
      has_conflicts = false,
      relpath = "minimal.lua",
      repo = {
        abbrev_head = "main",
        gitdir = "C:/Users/Cyprien/AppData/Local/nvim/.git",
        toplevel = "C:/Users/Cyprien/AppData/Local/nvim",
        username = "vzytoi"
      }
    },
    gitdir_watcher = <userdata 1>,
    hunks = { {
        added = {
          count = 42,
          lines = { "vim.cmd [[set runtimepath=$VIMRUNTIME]]\r", "vim.cmd [[set packpath=/tmp/nvim/site]]\r", "local package_root = '/tmp/nvim/site/pack'\r", "local install_path = package_root .. '/packer/start/packer.nvim'\r", "\r", "if vim.fn.isdirectory(install_path) ==
0 then\r", '  print("Installing Gitsigns and dependencies.")\r', "  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }\r", "end\r", "\r", "-- Load plugins\r", "require('packer').startup {\r", "  {\r", "    'wbthomason/packer.nvim',\r", "    {\r", "      'lewis6991/gitsigns.nvim',\r", "      requires = { 'nvim-lua/plenary.nvim' },\r", "    },\r", "    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "  },\r", "  config = {\r", "    package_root = package_root,\r", "    compile_path = install_path .. '/plugin/packer_compiled.lua',\r", "    display = { non_interactive = true },\r", "  },\r", "}\r", "\r", "_G.load_config = function()\r", "  require('gitsigns').setup{\r", "    debug_mode = true, -- You must add this to enable debug messages\r", "    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "  }\r", "\r", "  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE\r", "\r", "  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES\r", "  vim.cmd('Gitsigns debug_messages')\r", "end\r", "\r", "require('packer').sync()\r", "\r", 'vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]\r' },
          start = 1
        },
        dend = 42,
        head = "@@ -0 +1,42 @@",
        removed = {
          count = 0,
          lines = {},
          start = 0
        },
        start = 1,
        type = "add",
        vend = 42
      } },
    pending_signs = {},
    staged_diffs = {}
  } }
```

**Comment by lewis6991** [comment]
> But my issue still occures, I guess it wasn't the reason.

Are you saying you modified Gitsigns to add the `--no-pager` option and it didn't fix the problem?

And to confirm `git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua` outputs the file contents?

**Comment by vzytoi** [comment]
Yes, it correctly output the file content.
input : ` git --no-pager --git-dir=C:/Users/Cyprien/AppData/Local/nvim/.git show :0:minimal.lua`
output : 
```
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'

if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Gitsigns and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end

-- Load plugins
require('packer').startup {
  {
    'wbthomason/packer.nvim',
    {
      'lewis6991/gitsigns.nvim',
      requires = { 'nvim-lua/plenary.nvim' },
    },
    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  },
  config = {
    package_root = package_root,
    compile_path = install_path .. '/plugin/packer_compiled.lua',
    display = { non_interactive = true },
  },
}

_G.load_config = function()
  require('gitsigns').setup{
    debug_mode = true, -- You must add this to enable debug messages
    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  }

  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE

  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES
  vim.cmd('Gitsigns debug_messages')
end

require('packer').sync()

vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by lewis6991** [comment]
Ok, so we know the `git show` command does work. That's something.

Couple of things to try.

Make sure we can run subprocesses.

Open a file named `test.lua` and add the contents:

```lua
require('gitsigns.subprocess').run_job({
  command = 'echo',
  args = {'hello'}
}, function(_, _, stdout, _)
    print(stdout)
end)
```

Then open in Neovim and run `:source % | messages`

You should see `hello` in the output.

Next run the same thing but run `git show`:

```lua
local gitdir = 'C:/Users/Cyprien/AppData/Local/nvim/.git'
local gitobj = ':0:minimal.lua'

require('gitsigns.subprocess').run_job({
  command = 'git',
  args = {
    '--no-pager',
    '--git-dir='..gitdir,
    'show',
    gitobj,
  }
}, function(code, signal, stdout, stderr)
    print('CODE: '..code)
    print('SIGNAL: '..signal)
    if stderr then print('STDERR: '..stderr) end
    if stdout then print('STDOUT: '..stdout) end
end)
```

What output do you get?

**Comment by vzytoi** [comment]
Thanks for your feed back. 
I used this configuration : 

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'

if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Gitsigns and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end

-- Load plugins
require('packer').startup {
  {
    'wbthomason/packer.nvim',
    {
      'lewis6991/gitsigns.nvim',
      requires = { 'nvim-lua/plenary.nvim' },
    },
    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  },
  config = {
    package_root = package_root,
    compile_path = install_path .. '/plugin/packer_compiled.lua',
    display = { non_interactive = true },
  },
}

_G.load_config = function()
  require('gitsigns').setup{
    debug_mode = true, -- You must add this to enable debug messages
    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  }

  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE

  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES
  vim.cmd('Gitsigns debug_messages')
end

require('packer').sync()

vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]

require('gitsigns.subprocess').run_job({
  command = 'echo',
  args = {'hello'}
}, function(_, _, stdout, _)
    print(stdout)
end)

local gitdir = 'C:/Users/Cyprien/AppData/Local/nvim/.git'
local gitobj = ':0:minimal.lua'

require('gitsigns.subprocess').run_job({
  command = 'git',
  args = {
    '--no-pager',
    '--git-dir='..gitdir,
    'show',
    gitobj,
  }
}, function(code, signal, stdout, stderr)
    print('CODE: '..code)
    print('SIGNAL: '..signal)
    if stderr then print('STDERR: '..stderr) end
    if stdout then print('STDOUT: '..stdout) end
end)
```

And got this output : 

```
hello
CODE: 0
SIGNAL: 0
STDOUT: vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Gitsigns and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
-- Load plugins
require('packer').startup {
  {
    'wbthomason/packer.nvim',
    {
      'lewis6991/gitsigns.nvim',
      requires = { 'nvim-lua/plenary.nvim' },
    },
    -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  },
  config = {
    package_root = package_root,
    compile_path = install_path .. '/plugin/packer_compiled.lua',
    display = { non_interactive = true },
  },
}
_G.load_config = function()
  require('gitsigns').setup{
    debug_mode = true, -- You must add this to enable debug messages
    -- GITSIGNS CONFIG THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  }
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  -- ALSO INCLUDE GITSIGNS DEBUG MESSAGES
  vim.cmd('Gitsigns debug_messages')
end
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```
Have a good day! (:

**Comment by lewis6991** [comment]
Ok so it seems the git show is working but isn't being placed in the cache for whatever reason.

You'll have to debug this yourself.

The main line to look it is [`teal/gitsigns/manager.tl:243`](https://github.com/lewis6991/gitsigns.nvim/blob/main/teal/gitsigns/manager.tl#L243)

```lua
  if not bcache.compare_text or config._refresh_staged_on_update then
    bcache.compare_text = git_obj.repo:get_show_text(compare_object)
  end
```

and [`teal/gitsigns/git.tl:246`](https://github.com/lewis6991/gitsigns.nvim/blob/main/teal/gitsigns/git.tl#L246):

```lua
--- Run git command the with the objects gitdir and toplevel
Repo.command = function(self: Repo, args: {string}, spec: GJobSpec): {string}, string
  spec = spec or {}
  spec.cwd = self.toplevel
  return M.command({'--git-dir='..self.gitdir, unpack(args)}, spec)
end

...

Repo.get_show_text = function(self: Repo, object: string): {string}, string
  return self:command({'show', object}, {supress_stderr = true})
end
```

**Comment by fedepujol** [comment]
Was getting this behaviour on my Windows machine with Neovim nightly but, downgrading to 0.5.1 seems to fix the issue.

Could this be related to something being introduce in the newer versions?

Will try to debug, since I've updated to neovim 0.6, and this error occurs.

**Comment by lewis6991** [comment]
The biggest different between 0.5.* and 0.6+ is the inclusion of `vim.diff`.

Can you try:

```lua
require('gitsigns').setup{
  ... -- your config
  diff_opts = {
    internal = false
  }
}
```

This will cause Gitsigns to use `git diff` via a subprocess as opposed to using `vim.diff` internally.

I'm thinking this might just be a CRLF issue.

**Comment by Sh3Rm4n** [comment]
I have the same issue and the point, that something between 0.5.* and 0.6+ has changed, is a good hint. I update neovim nightly regularly and have noticed this issue coming up sometime ago, but did not remember when. This is why I failed to bisect this issue in https://github.com/lewis6991/gitsigns.nvim/issues/410, because I did not change the neovim version. 

I can confirm, that turning off the internal diff mode does fix the issue for me.

**Comment by fedepujol** [comment]
Toggling off the internal diff fixes my issue but, changing the `fileformat` to `unix` does solve this as well (without turning off the internal diff).

What would be the best solution?

**Comment by lewis6991** [comment]
Can you give #417 a try? Not sure if it will fix the issue but worth a shot.

**Comment by fedepujol** [comment]
I try #417 with `fileformat=dos` (without turning off internal diff) and it fixes the issue on my Windows machine.

**Comment by vzytoi** [comment]
> ```lua
>   diff_opts = {
>     internal = false
>   }
> ```

I can confirm that turning this off solved it for me too.

**Comment by fedepujol** [comment]
I've tested #417 in a Linux machine with `ff=dos` and the issue persisted. But if you stripped the `\r` of the lines instead of adding one, the issue is resolved.

Here is the solution I've got it to work:
```lua
local function get_lines(bufnr)

   local buftext = api.nvim_buf_get_lines(bufnr, 0, -1, false)
   if util.is_unix and api.nvim_buf_get_option(bufnr, 'fileformat') == 'dos' then
      for i = 1, #buftext do
         buftext[i] = string.gsub(buftext[i], '\r', '')
      end
   end
   return buftext
end
```
Hope it helps!

**Comment by lewis6991** [comment]
Well that's interesting as such a change would make the issue appear for me.

This tells me that your output from `git show` is stripping `\r`. Would you be able to confirm this is the case? If so the fix is likely to pass an extra option to `git show`.

**Comment by lewis6991** [comment]
I've updated #417 to set `core.autocrlf=false` instead. Maybe that fixes the issue?

**Comment by lewis6991** [comment]
Ok worked on this some more. Can people give #417 another try?

**Comment by dineshKumar777** [comment]
> Ok worked on this some more. Can people give #417 another try?

I am having the same issue in my windows machine. #417 resolved it for me.

![image](https://user-images.githubusercontent.com/21097695/145287152-d9aa5698-baf2-473a-afc2-bfd54608f4ba.png)

**Comment by lewis6991** [comment]
Should be fixed in #417
